### ISSUE TITLE
Add a rule: meta charset/encoding declaration must be within the first 1024 bytes

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3968]
## Summary

markuplint has no rule checking that a `<meta charset>` (or `<meta http-equiv="content-type">` encoding declaration) is serialized within the first 1024 *bytes* of the document, as HTML LS requires. No byte-offset-based check exists anywhere in the codebase today.

## Repro

```html
<!doctype html><title>...</title><!-- ~1000+ chars of padding --><meta charset=utf-8>
```

nu-validator: error — `A "charset" attribute on a "meta" element found after the first 1024 bytes.`

markuplint (all bench rules enabled): 0 violations.

Fixture: `tests/external/validator/tests/html/parser/charset-after-1024-novalid.html` (`nu-only`).

## Spec basis

[HTML LS §4.2.5.4 "Specifying the document's character encoding"](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-charset) (paraphrased location — verify exact section number against the live spec before implementing, per this repo's doc-accuracy policy):

> The element containing the character encoding declaration must be serialized completely within the first 1024 bytes of the document.

This is a `must` requirement, distinct from and additional to the existing `charset` enum/format checks in `packages/@markuplint/html-spec/src/spec.meta.jsonc`.

## Why this doesn't fit the existing type-check infrastructure

Every existing HTML LS byte/offset-sensitive rule in this codebase operates on parsed-node structure (element order, attribute values, ancestor/sibling relationships) — none currently measure a *byte offset from the start of the document*. `@markuplint/ml-ast` node positions (`startOffset` etc., `packages/@markuplint/ml-ast/src/types.ts`) are in UTF-16 code units (JS string offsets), not bytes, and the document's actual serialized byte length depends on its encoding (typically UTF-8, where the boundary between 1-byte and multi-byte characters matters for an exact 1024-byte cutoff).

## Proposed direction

New rule (e.g. `charset-declaration-within-1024-bytes` or similar name — TBD at implementation time) that:

1. Locates the `<meta charset>` / `<meta http-equiv="content-type">` encoding-declaration element (if any).
2. Computes the UTF-8 byte length of the document's raw source from offset 0 up to and including that element's end (`el.endOffset`), being careful with surrogate pairs when converting UTF-16 code-unit offsets to UTF-8 byte counts (e.g. `Buffer.byteLength(document.raw.slice(0, el.endOffset), 'utf8')` in Node, or an equivalent encoding-aware calculation that works across the project's supported runtimes — see `packages/@markuplint/shared` cross-runtime constraints).
3. Reports a violation if that byte length exceeds 1024.

This is a narrower, more mechanical rule than most — it has no "content model" or "attribute reference" shape, just a byte-position assertion — so it may warrant its own category or at minimum careful README documentation of the byte-vs-code-unit distinction.

## Files

- `packages/@markuplint/rules/src/` (new rule directory)
- `packages/@markuplint/rules/src/index.ts`
- `packages/@markuplint/rules/schema.json`
- `packages/@markuplint/config-presets/src/preset.html-standard.jsonc`
- `tests/external/bench/config.ts` (enable once implemented)

## Bench impact

1 fixture currently `nu-only`. Resolving this Issue is expected to flip it to `match-error`.

**Comment by YusukeHirao** [comment]
Duplicate of #3943, which already tracks this fixture (charset-after-1024) alongside 4 related parse5-gap fixtures. Closing in favor of the existing Issue.
